src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -920,33 +919,24 @@ int32_t NaClSysRead(struct NaClAppThread  *natp,
     count = INT32_MAX;
   }
 
-  NaClVmIoWillStart(nap,
-                    (uint32_t) (uintptr_t) buf,
-                    (uint32_t) (((uintptr_t) buf) + count - 1));
   read_result = ((struct NaClDescVtbl const *)ndp->base.vtbl)->Read(ndp, (void *)sysaddr, count);
 
-  NaClVmIoHasEnded(nap,
-                    (uint32_t) (uintptr_t) buf,
-                    (uint32_t) (((uintptr_t) buf) + count - 1));
```

Comment:
I actually don't think we can remove these. What these do is mark which memory addresses are currently being written to/read to in order to prevent them from being mmapped/mprotected over. I'm pretty sure that removing these would allow the user to write to memory they should not be able to under some conditions.